pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/ZookeeperResource.java

```diff
@@ -187,14 +181,47 @@ public String ls(
 
     path = validateAndNormalizeZKPath(path, true);
 
-    List<String> children = _pinotHelixResourceManager.getZKChildren(path);
+    List<String> children = _pinotHelixResourceManager.getZKChildNames(path);
     try {
       return JsonUtils.objectToString(children);
     } catch (JsonProcessingException e) {
       throw new RuntimeException(e);
     }
   }
 
+  @GET
+  @Path("/zk/getChildren")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Get all child znodes")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"),
+      @ApiResponse(code = 404, message = "ZK Path not found"),
+      @ApiResponse(code = 204, message = "No Content"),
+      @ApiResponse(code = 500, message = "Internal server error")
+  })
+  public String getChildren(
+      @ApiParam(value = "Zookeeper Path, must start with /", required = true) @QueryParam("path") String path) {
+
+    path = validateAndNormalizeZKPath(path, true);
+
+    List<ZNRecord> znRecords = _pinotHelixResourceManager.getZKChildren(path);
+    if (znRecords != null) {
+      List<String> resp = new ArrayList<>();
+      for (ZNRecord znRecord : znRecords) {
+        if (znRecord != null) {
+          byte[] serializeBytes = serialize(znRecord);
+          resp.add(new String(serializeBytes, StandardCharsets.UTF_8));
+        }
+      }
+      try {
+        return JsonUtils.objectToString(resp);
```

Comment:
This will cause `ZNRecord` encoded as escaped string. Try `JsonUtils.objectToPrettyString(znRecords)`

Reply:
I think this is the current behavior of the `zk/get` return as well (e.g. also escaped).